### PR TITLE
Fixed semantic versioning check and added unit tests

### PR DESCRIPTION
**Pull request by bonniecpk**
#32

### CODE REVIEW

File: oauthdoctor/diag/config.go [hunk 1]
```diff
@@ -48,12 +48,14 @@ const (
...
+var (
+	// PIIWords is a slice of constant strings that indicate Personally Identifiable Information
+	PIIWords = []string{DevToken, ClientID, ClientSecret, RefreshToken}
 
-// RequiredKeys are the key names used in the Language structure that defines
-// the contents of a client library configuration file.
-var RequiredKeys = []string{DevToken, ClientID, ClientSecret, RefreshToken}
+	// RequiredKeys are the key names used in the Language structure that defines
+	// the contents of a client library configuration file.
+	RequiredKeys = []string{DevToken, ClientID, ClientSecret, RefreshToken}
```
Comment (bobhancock) [inline oauthdoctor/diag/config.go]: If these contain the same constants, do we need two slices just with different names.

File: oauthdoctor/diag/config.go [hunk 2]
```diff
@@ -536,15 +538,23 @@ func (c *ConfigFile) Validate() (bool, error) {
 
 // MinGoVersion test for the minimum version of Go required.
```
Comment (bobhancock) [inline oauthdoctor/diag/config.go]: Add  test saying that we require a minimum of 1.11 because we are using modules and include a link to the Wiki.

File: oauthdoctor/diag/config.go [hunk 3]
```diff
@@ -536,15 +538,23 @@ func (c *ConfigFile) Validate() (bool, error) {
...
 	}
-	if v < min {
-		return fmt.Errorf("minimum required Go version is 1.11: you are running %s", ver)
+	if num < min {
+		return fmt.Errorf("minimum required Go version is 1.11: you are running %s", v)
 	}
 	return nil
 }
+
+var semanticVersionRe = regexp.MustCompile(`^[0-9\.]+`)
```
Comment (bobhancock) [inline oauthdoctor/diag/config.go]: Consider a var name of semanticVersionRegex to make it clearer.

File: oauthdoctor/diag/config_test.go [hunk 4]
```diff
@@ -489,6 +489,59 @@ func TestParseXMLFile(t *testing.T) {
...
+			desc:    "Version 1.rc is not supported",
+			version: "1.rc",
+			wantStr: "minimum required",
+		},
+	}
+
+	for _, test := range tests {
+		got := checkGoVersion(test.version)
+
+		if got != test.want && !strings.Contains(got.Error(), test.wantStr) {
```
Comment (bobhancock) [inline oauthdoctor/diag/config_test.go]: Why the compound condition?

File: oauthdoctor/diag/config.go (hunk 1 above)
Comment (bonniecpk) [reply oauthdoctor/diag/config.go]: The elements in the slices just happen to be the same. They can diverge at some point when more config keys are added.

File: oauthdoctor/diag/config.go (hunk 2 above)
Comment (bonniecpk) [reply oauthdoctor/diag/config.go]: Do you mean unit tests? It's already added.

File: oauthdoctor/diag/config_test.go (hunk 4 above)
Comment (bonniecpk) [reply oauthdoctor/diag/config_test.go]: Fixed

File: oauthdoctor/diag/config.go (hunk 2 above)
Comment (bobhancock) [reply oauthdoctor/diag/config.go]: Fat fingers. That should be text.
